Pod/Classes/SKYContainer.m

```diff
@@ -642,6 +653,29 @@ - (void)registerDeviceCompletionHandler:(void (^)(NSString *, NSError *))complet
                               }];
 }
 
+- (void)unregisterDevice
+{
+    [self unregisterDeviceCompletionHandler:^(NSString *deviceID, NSError *error) {
+        if (error != nil) {
+            NSLog(@"Warning: Fail to unregister device: %@", error.localizedDescription);
+            return;
+        }
+    }];
+}
+
+- (void)unregisterDeviceCompletionHandler:(void (^)(NSString *deviceID,
+                                                    NSError *error))completionHandler
+{
+    NSString *existingDeviceID = self.registeredDeviceID;
+    if (existingDeviceID != nil) {
+        SKYUnregisterDeviceOperation *operation =
+            [SKYUnregisterDeviceOperation operationWithDeviceID:existingDeviceID];
+        operation.unregisterCompletionBlock = completionHandler;
```

Comment:
The `completionHandler` has to be called in the main thread. (do not fix it in the Operation)